commands/create.go

```diff
@@ -458,3 +480,101 @@ func tlsPath(c CommandLine, flag string, defaultName string) string {
 
 	return filepath.Join(mcndirs.GetMachineCertDir(), defaultName)
 }
+
+func updateUserdataFile(driverOpts *rpcdriver.RPCFlags, userdataFlag, customInstallScript string) error {
+	var userdataContent []byte
+	var err error
+	userdataFile := driverOpts.String(userdataFlag)
+
+	if userdataFile == "" {
+		if !strings.HasSuffix(userdataFlag, "cloud-config") {
+			// If the userdata file is not set, then the customInstallScript file is set as the userdata file.
+			driverOpts.Values[userdataFlag] = customInstallScript
```

Comment:
doesn't this get overwritten below ?

Reply:
No, the next line is a `return` statement.